driver/normalizer/annotation.go

```diff
@@ -1,34 +1,368 @@
+			"kind": {Op: Var("kind")},
+		},
+		LookupArrOpVar("kind", litExprRoles),
+	),
+
+	AnnotateType("CPPASTCompoundStatement", nil, role.Body),
+	AnnotateType("CPPASTDeclarator", FieldRoles{
+		"Name": {Rename: uast.KeyToken},
+	}, role.Declaration, role.Variable, role.Name),
+	AnnotateType("CPPASTDeclarator", nil, role.Declaration, role.Variable, role.Name),
+
+	AnnotateType("CPPASTFunctionDefinition", ObjRoles{
+		"Prop_Body": {role.Function, role.Declaration, role.Body},
+		"Prop_DeclSpecifier": {role.Function, role.Declaration, role.Return, role.Type},
+	}, role.Function, role.Declaration),
+
+	AnnotateType("CPPASTFunctionDeclarator", FieldRoles{
+		"Prop_Name": {Roles: role.Roles{role.Function, role.Declaration, role.Name}},
+		// SDK TODO: adding "Opt: true" fails since Arrays can't be optional, but without it the annotation won't
+		// match, thus the duplicated annotation below
+		"Prop_Parameters": {Arr: true, Roles: role.Roles{role.Function, role.Declaration, role.Argument}},
+	}, role.Function, role.Declaration),
+
+	AnnotateType("CPPASTFunctionDeclarator", FieldRoles{
+		"Prop_Name": {Roles: role.Roles{role.Function, role.Declaration, role.Name}},
+	}, role.Function, role.Declaration),
+
+	AnnotateType("CPPASTReturnStatement", ObjRoles{
+		"Prop_ReturnArgument": {role.Return, role.Value},
+	}, role.Statement, role.Return),
+
+	AnnotateTypeCustom("CPPASTBinaryExpression", MapObj(Obj{
+		"Operator": Var("operator"),
+		"Prop_Operand1": ObjectRoles("operand1"),
+		"Prop_Operand2": ObjectRoles("operand2"),
+		// Temporarily using the same name to detect if those are really duplicated
+		"Prop_InitOperand2": ObjectRoles("operand2"),
+	}, Obj{
+		uast.KeyToken: Var("operator"),
+		"Prop_Operand1": ObjectRoles("operand1", role.Binary, role.Expression, role.Left),
+		"Prop_Operand2": ObjectRoles("operand2", role.Binary, role.Expression, role.Right),
+	}), LookupArrOpVar("operator", binaryExprRoles)),
+
+	AnnotateType("CPPASTEqualsInitializer", nil, role.Declaration, role.Assignment, role.Expression, role.Right),
+
+	AnnotateType("CPPASTCompositeTypeSpecifier", FieldRoles{
+		"Key": {Op: String("struct")},
+	} , role.Declaration, role.Type),
+
+	AnnotateType("CPPASTCompositeTypeSpecifier", FieldRoles{
+		"Key": {Op: String("struct")},
+		"Prop_Members": {Arr: true, Roles: role.Roles{role.Declaration, role.Type}},
+	} , role.Declaration, role.Type),
+
+	AnnotateType("CPPASTElaboratedTypeSpecifier", FieldRoles{
+		"Kind": {Op: String("enum")},
+	} , role.Declaration, role.Type, role.Enumeration),
+
+	AnnotateType("CPPASTCompositeTypeSpecifier", FieldRoles{
+		"Key": {Op: String("class")},
+	} , role.Declaration, role.Type),
+
+	// No Union role
+	AnnotateType("CPPASTCompositeTypeSpecifier", FieldRoles{
+		"Key": {Op: String("union")},
+		"Prop_Members": {Arr: true, Roles: role.Roles{role.Declaration, role.Type, role.Incomplete}},
+		//"Prop_Clauses": {Arr: true, Roles: role.Roles{role.Declaration, role.Type, role.Incomplete}},
```

Comment:
Remove or add a explanation why it's commented out

Reply:
Duplicated roles with variable arrays that weren't actually duplicated after it. Good catch.